models/edge_detection_dexined/edge_detection.py

```diff
@@ -0,0 +1,79 @@
+'''
+This sample demonstrates edge detection with dexined (DNN based) edge detection techniques.
+'''
+import os
+import cv2 as cv
+import argparse
+import numpy as np
+
+def get_args_parser(func_args):
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument('--input', help='Path to input image or video file. Skip this argument to capture frames from a camera.', default=0, required=False)
+    parser.add_argument('--model', help='Path to dexined.onnnx', default='edge_detection_dexined_2024sep.onnx', required=False)
+
+
+    args, _ = parser.parse_known_args()
+    parser = argparse.ArgumentParser(parents=[parser],
+                                     description='', formatter_class=argparse.RawTextHelpFormatter)
+    return parser.parse_args(func_args)
+
+def sigmoid(x):
+    return 1.0 / (1.0 + np.exp(-x))
+
+def post_processing(output, shape):
+    h, w = shape
+    preds = []
+    for p in output:
+        img = sigmoid(p)
+        img = np.squeeze(img)
+        img = cv.normalize(img, None, 0, 255, cv.NORM_MINMAX, cv.CV_8U)
+        img = cv.resize(img, (w, h))
+        preds.append(img)
+    fuse = preds[-1]
+    ave = np.array(preds, dtype=np.float32)
+    ave = np.uint8(np.mean(ave, axis=0))
+    return fuse, ave
+
+def loadModel(args):
```

Comment:
Could you write a wrapper class and put it in a separate file? We do this in this way so that we can run benchmark on different models.

Example: https://github.com/opencv/opencv_zoo/blob/main/models/face_detection_yunet/yunet.py

Reply:
added a separate file with class for python and added a class in edge_detection.cpp